phoenix-core/src/main/java/org/apache/phoenix/util/IndexUtil.java

```diff
@@ -849,4 +855,35 @@ public static long getIndexPendingDisableCount(PhoenixConnection conn, String fa
             throw new IOException(e);
         }
     }
+
+    public static void setScanAttributes(Scan scan, PTable table, PhoenixConnection phoenixConnection) throws SQLException {
+        if (table.isTransactional() || table.isImmutableRows() || table.getType() != PTableType.INDEX) {
```

Comment:
nit: I feel it would be better to extract this and other noop logic outside of the function as it doesn't set anything as the name suggests.

Reply:
If I do that, then I need to do these checks before calling the function. Now, if the function is called in multiple places, then I need to repeat those checks. This would be error prone. Also, how do I make sure/specify for future invocations those checks will happen before this function is called?

Reply:
@kadirozde: How about having a boolean class variables? For a given table/query, things will not change, until the transaction is complete.

Reply:
So you suggest adding tens of lines of code for removing couple lines of code. That does not sound good. I really do not understand why the if statement for checking the input arguments is an issue here.

Reply:
It's a nit anyway. Feel free to ignore if you'd like. I would do it to avoid making the function a misnomer (it says set but doesn't really set anything in some cases).